src/Microsoft.AspNet.Hosting/Startup/FailSafeStartupLoader.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+using Microsoft.Dnx.Runtime;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.Logging;
+
+namespace Microsoft.AspNet.Hosting.Startup
+{
+    /// <summary>
+    /// Catches common startup exceptions and shows an error page.
+    /// </summary>
+    public class FailSafeStartupLoader : IStartupLoader
```

Comment:
I'm going to +1, this feels like a larger than needs to be change. If your current tests are covering all of the cases, we can restructure to be a simple try catch in the right places (we can make things happen in a single place). Rather than using the startup loader.